### PR TITLE
tests: add more real world test case of ingest

### PR DESCRIPTION
**Pull request by choleraehyq**
@siddontang @zhangjinpeng1987 PTAL
Signed-off-by: Cholerae Hu <huyingqian@pingcap.com>

### CODE REVIEW

File: tests/test_ingest_external_file.rs [hunk 1]
```diff
@@ -246,3 +246,105 @@ fn test_ingest_external_file_new_cf() {
     assert_eq!(snap.get_cf(handle, b"k2").unwrap().unwrap(), b"b");
     assert_eq!(snap.get_cf(handle, b"k3").unwrap().unwrap(), b"c");
 }
+
+fn check_kv(db: &DB, cf: Option<&CFHandle>, data: &[(&[u8], Option<&[u8]>)]) {
+    for &(k, v) in data {
+        if cf.is_some() {
+            if v.is_none() {
```
Comment (siddontang) [inline tests/test_ingest_external_file.rs]: can we use `assert_eq!(db.get_cf(cf.unwrap(), k).unwrap(), v);` directly?

File: tests/test_ingest_external_file.rs [hunk 2]
```diff
@@ -246,3 +246,105 @@ fn test_ingest_external_file_new_cf() {
...
+        iter.next();
+    }
+    writer.finish().unwrap();
+}
+
+#[test]
+fn test_ingest_simulate_real_world() {
+    const ALL_CFS: [&str; 3] = ["lock", "write", "default"];
+    let path = TempDir::new("_rust_rocksdb_ingest_real_world_1").expect("");
+    let path_str = path.path().to_str().unwrap();
```
Comment (siddontang) [inline tests/test_ingest_external_file.rs]: seem you can extract a function to create db1 and db2.

File: tests/test_ingest_external_file.rs (hunk 1 above)
Comment (choleraehyq) [inline tests/test_ingest_external_file.rs]: No, `==` cannot apply on `Option`.

Comment (siddontang) [inline tests/test_ingest_external_file.rs]: if cf is none, we can use default CFHandle, and the logic may look simpler.

Comment (choleraehyq) [inline tests/test_ingest_external_file.rs]: Done.

File: tests/test_ingest_external_file.rs [hunk 3]
```diff
@@ -246,3 +239,105 @@ fn test_ingest_external_file_new_cf() {
...
+}
+
+#[test]
+fn test_ingest_simulate_real_world() {
+    const ALL_CFS: [&str; 3] = ["lock", "write", "default"];
+    let path = TempDir::new("_rust_rocksdb_ingest_real_world_1").expect("");
+    let mut db = create_default_database(&path);
+    let gen_path = TempDir::new("_rust_rocksdb_ingest_real_world_new_cf").expect("");
+
+    for cf in &ALL_CFS {
```
Comment (siddontang) [inline tests/test_ingest_external_file.rs]: you can extract create cf too.

**Comment by siddontang** [comment]
LGTM
PTAL @zhangjinpeng1987

**Pull request merged**